Epsilon.Abstractions/Components/AbstractCompetenceComponent.cs

```diff
@@ -1,18 +1,77 @@
-﻿using DocumentFormat.OpenXml.Packaging;
+﻿using DocumentFormat.OpenXml;
+using DocumentFormat.OpenXml.Packaging;
 using DocumentFormat.OpenXml.Wordprocessing;
 
 namespace Epsilon.Abstractions.Components;
 
 public abstract class AbstractCompetenceComponent: IWordCompetenceComponent
 {
-    protected AbstractCompetenceComponent(IAsyncEnumerable<LearningDomainSubmission> submissions, IEnumerable<LearningDomain?> domains)
+    protected AbstractCompetenceComponent(IAsyncEnumerable<LearningDomainSubmission> submissions, IEnumerable<LearningDomain?> domains, IEnumerable<LearningDomainOutcome> outcomes)
     {
         Submissions = submissions;
         Domains = domains;
+        Outcomes = outcomes;
     }
 
     protected IAsyncEnumerable<LearningDomainSubmission> Submissions { get; set; }
     protected IEnumerable<LearningDomain?> Domains { get; set; }
+    
+    protected IEnumerable<LearningDomainOutcome> Outcomes { get; set; }
 
-    public abstract Task<Body> AddToWordDocument(MainDocumentPart mainDocumentPart);
+    public abstract Task<Body?> AddToWordDocument(MainDocumentPart mainDocumentPart);
+
+    protected static Paragraph FormattedText(string text)
```

Comment:
The general functions are great, 
Just change the name so it indicates what the function does. 
In most cases now it creates something. But can be miss read that things are updates or something else.